src/utilities/createRouterRoute.ts

```diff
@@ -1,15 +1,24 @@
 import { reactive } from 'vue'
-import { ResolvedRoute } from '@/types'
+import { Route } from '@/types'
+import { RouterRouteQuery } from '@/utilities/createRouterRouteQuery'
 
-type RouterRouteUpdate = (matched: ResolvedRoute) => void
+export type RouterRoute = {
+  matched: Route,
+  matches: Route[],
+  name: string,
+  query: RouterRouteQuery,
+  params: Record<string, unknown>,
+}
+
+type RouterRouteUpdate = (route: RouterRoute) => void
 
-type RouterRoute = {
-  route: ResolvedRoute,
+type CreateRouterRoute = {
+  route: RouterRoute,
   updateRoute: RouterRouteUpdate,
 }
 
-export function createRouterRoute(fallbackRoute: ResolvedRoute): RouterRoute {
-  const route = reactive<ResolvedRoute>(fallbackRoute)
+export function createRouterRoute(fallbackRoute: RouterRoute): CreateRouterRoute {
```

Comment:
if we're going to have a thing called `RouterRoute`, I think we should rename this file & function. Perhaps `createCurrentRoute` or `createReactiveRoute`.

Reply:
how is this any different than `createRouterPush` or the other factory methods used for creating the router object?

Reply:
because `RouterRoute` is a thing, when I see a function `createRouterRoute` I expect it to return `RouteRoute`, not take `RouterRoute` as an arg and return `CreateRouterRoute`. Our naming convention is causing conflicts here

Reply:
I mean it does return a RouterRoute and a method to update that route. So I don't see any conflict here. But we could change our naming convention away from creatRouterX to something like routerXFactory which would leave createX available for the use case you want. 

But we've already talked about swapping RouterRoute with ResolvedRoute so does that help at all?

Either way, the code is solid yeah? Ultimately we're not gonna be able to name everything perfect. And id like our router.thing to all follow the same naming convention. So unless we're changing that convention IMO this should be called createRouterRoute. 

But I'd be fine with any other convention as long as we change them all.

Reply:
yes the code is solid. I think these discussions over naming are worthwhile, despite how tedious it may seem. This function returns a thing called `CreateRouterRoute`, so following our convention of methods like `createRouterPush`, `createRouterFind`, etc this should be `createCreateRouterRoute` as ugly as that is. Perhaps there's a better term for this thing? What do you think of naming the type `RouterCurrentRoute` and the method `createRouterCurrentRoute`? or `RouterActiveRoute`, something like that?

Reply:
The current pattern is all of the properties of `Router` are made in factories following the `createRouterX` pattern where `X` is the router property. Some of those factories return multiple values to support the primary `X` being created. They return `X` directly OR `X` is one of the values it returns. Both `createRouterRoute` and `createRouterReject` return `CreateRouterX` because they return more than just the router property they're responsible for creating. Which all makes sense to me. 

If you feel strongly that any `createX` method should return `X` and only `X` then we need a different naming convention for these factory methods. I suggested `routerXFactory` (`routerRouteFactory`, `routerPushFactory`, etc) where `X` is the router property NOT the type being returned. Do you have any thoughts on that suggestion?